### DISCUSSION TITLE
Error in Step 7 of ICSE processing

### DISCUSSION THREAD

**Discussion by PriyomRoy** [alexisInSAR/EZ-InSAR#56] [Help]
I am encountering a error at step 7 of ICSE processing as follows:
-----------------------------------------------------------------------------------------------------------
IndexError: list index out of range
Completed Parsing the Configuration file
Functions to be executed:
['Function-1', 'Function-2', 'Function-3', 'Function-4']
Running: generateIgram
['--reference', '/matlab/joshi_des/coreg_secondarys/20230512', '--secondary', '/matlab/joshi_des/coreg_secondarys/20230524', '--interferogram', '/matlab/joshi_des/coarse_interferograms/20230512_20230524', '--interferogram_prefix', 'int', '--overlap']
Traceback (most recent call last):
  File "/home/nrsc/tools_insar/proc_insar/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 251, in <module>
    main(args.start,args.end)
  File "/home/nrsc/tools_insar/proc_insar/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 242, in main
    cfgParser.runCmd()
  File "/home/nrsc/tools_insar/proc_insar/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 54, in runCmd
    func_modules.main(self.funcParams[section])
  File "/home/nrsc/tools_insar/proc_insar/isce2/contrib/stack/topsStack/generateIgram.py", line 163, in main
    minReference = topReference.bursts[0].burstNumber
                   ~~~~~~~~~~~~~~~~~~~^^^
  File "/home/nrsc/mconda/miniconda3/envs/InSARenv/lib/python3.11/site-packages/isce/components/iscesys/Component/TraitSeq.py", line 358, in __getitem__
    return self.list[i]
------------------------------------------------------------------------------------------------------------------------

not sure why it is happening. All previous steps are fine.
My study area is splitting across two frames as seen in attachment. Is that the reason?
![study area error](https://github.com/alexisInSAR/EZ-InSAR/assets/132996953/80c4de6b-57fc-451d-8148-194d624e2d90)

**Comment by alexisInSAR** [comment]
Hi, 

There seems to be a date missing... Can you check your previous files? 

Kind regards, 

Alexis

**Reply by PriyomRoy** [reply]
I cross checked the slc.list file versus the number of downloads. Both show 204 scenes. The Download SLC option of EZ-Insar shows all files are downloaded. The check IPF option is running fine. Error persists in ICSE processing. Can figure out what's wrong. 
The workflow has been tested with full run earlier for a different study area. It was fine.

**Reply by PriyomRoy** [reply]
*cant figure

**Reply by PriyomRoy** [reply]
This problem is solved. There were incomplete downloads

**Reply by whz132** [reply]
> This problem is solved. There were incomplete downloads

Hi, I'm having the same problem. But I'm getting this error on every pair. Is that what happened to you?

**Comment by PriyomRoy** [comment]
Error during ps_plot. SBAS and Merged plots are fine


Error in ps_plot (line 325)
xy=ps.xy;

**Reply by alexisInSAR** [reply]
Hi, 

Can you explain in details this error when using ps_plot with PS processing? 

Kind regards, 

Alexis

**Comment by PriyomRoy** [comment]
The following error is showing (see attached)

**Comment by PriyomRoy** [comment]
Hi,
I wish to flag an issue, not sure if any one else has faced it or it is
specific to me, follows:
Once the PS workflow is executed and then ps_plot command is tried, all the
plots are being generated and time series is also working fine. However,
post the PS workflow, if one executes the SBAS, upon successful execution,
a "SMALL BASELINE" sub folder is created. After this, if one attempts the
ps_plot from the master folded (e.g. "INSAR_20220318) to generate the ps
velocity, the following error is returned:
--------------------------------
>> ps_plot('v')
Unrecognized field name "xy".
Error in ps_plot (line 325)
xy=ps.xy;
--------------------------------

However, the command to generated SBAS velocity (i.e. ps_plot ('V')) from
within the  "SMALL BASELINE" sub folder is working fine.

On Sun, Jul 2, 2023 at 12:02 PM Priyom Roy ***@***.***> wrote:

> The following error is showing (see attached)
>
> On Sat, 1 Jul 2023, 22:55 Alexis Hrysiewicz, ***@***.***>
> wrote:
>
>> Hi,
>>
>> Can you explain in details this error when using ps_plot with PS
>> processing?
>>
>> Kind regards,
>>
>> Alexis
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/alexisInSAR/EZ-InSAR/discussions/56#discussioncomment-6333380>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A7WV6WKMBUL7QDRR3PYDZGLXOBMRXANCNFSM6AAAAAAZPY2D5I>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 

प्रियम रॉय Priyom Roy
वैज्ञानिक Scientist SE
भूविज्ञान ग्रुप Geosciences Group
राष्ट्रीय सुदूर संवेदन केन्द्र National Remote Sensing Centre
भारतीय अन्तरिक्ष अनुसंधान संगठन Indian Space Research Organisation
हैदराबाद, भारत Hyderabad, India
